File: src/musikdroid/app/src/main/java/io/casey/musikcube/remote/playback/RepeatMode.java
package io.casey.musikcube.remote.playback;

public enum RepeatMode {
    None("none"),
    List("list"),
    Track("track");

    private final String rawValue;

    RepeatMode(String rawValue) {
        this.rawValue = rawValue;
    }

    @Override
    public String toString() {
        return rawValue;
    }

    public static RepeatMode from(final String rawValue) {
        if (None.rawValue.equals(rawValue)) {
            return None;
        } else if (List.rawValue.equals(rawValue)) {
            return List;
        } else if (Track.rawValue.equals(rawValue)) {
            return Track;
        }

        throw new IllegalArgumentException("rawValue is invalid");
    }
}

File: src/musikdroid/app/src/main/java/io/casey/musikcube/remote/playback/RepeatMode.kt
package io.casey.musikcube.remote.playback

enum class RepeatMode constructor(private val rawValue: String) {
    None("none"),
    List("list"),
    Track("track");

    override fun toString(): String {
        return rawValue
    }

    companion object {
        fun from(rawValue: String): RepeatMode {
            if (None.rawValue == rawValue) {
                return None
            }
            else if (List.rawValue == rawValue) {
                return List
            }
            else if (Track.rawValue == rawValue) {
                return Track
            }

            throw IllegalArgumentException("rawValue matches invalid")
        }
    }
}
